SurveyAppTests/Tests/Services/Network/API/BaseAPITests.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  BaseAPITests.swift
+//  SurveyAppTests
+//
+//  Created by Chananchida F. on 6/17/21.
+//
+
+import Quick
+import Nimble
+import OHHTTPStubs
+
+@testable import SurveyApp
+
+enum Defines {
+
+    static let baseURL: String = "https://baseurl.com"
+}
+
+class BaseAPITests: QuickSpec {
+
+    override func spec() {
+        
+        let keychain = KeychainStorage.default
+        let userSession = UserSession(keychain: keychain)
+        var api: BaseAPI!
+        var configuration: RequestConfiguration!
+        
+        let sampleSuccessData = JSON.AuthenticationService.sampleUserCredential
+        let sampleFailureData = JSON.APIError.sampleAPIError
+        
+        describe("Test BaseAPI perform request") {
+            
+            context("when making a valid request") {
+                
+                beforeEach {
+                    api = BaseAPI(userSession: userSession)
+
+                    stub(condition: isMethodPOST()) { _ in
+                        HTTPStubsResponse(data: sampleSuccessData, statusCode: 200, headers: [:])
+                    }
+
+                    configuration = RequestConfiguration(
+                        method: .post,
+                        url: "\(Defines.baseURL)/test"
+                    )
+                }
```

Comment:
Create Mock object like and later you can use in test. Also you can out `api` init in `beforeEach` at a higher level so we have less duplicated code. Also we are using this https://github.com/krzysztofzablocki/Sourcery to help generate mocks. (It's a long read but will save you a lot of time in writing test.)
```

class UserSessionProviderProtocolMock: UserSessionProviderProtocol {

    var userSession: UserSessionProtocol?
}
```
```suggestion

            beforeEach {
                userSession = UserSessionProtocolMock()
                api = BaseAPI(userSession: userSession)
           }

            afterEach {
                Stubs.removeAll()
            }
            
            context("when making a valid request") {
                
                beforeEach {

                    stub(condition: isMethodPOST()) { _ in
                        HTTPStubsResponse(data: sampleSuccessData, statusCode: 200, headers: [:])
                    }

                    configuration = RequestConfiguration(
                        method: .post,
                        url: "\(Defines.baseURL)/test"
                    )
                }
```

Reply:
@blyscuit I updated the code please help check again 🙏